### DISCUSSION TITLE
Is there a way to remove duplicate feeds?

### DISCUSSION THREAD

**Discussion by github-yuzer** [FreshRSS/FreshRSS#3857] [Ideas]
I'd like to subscribe to 3 feeds from the same site, but sometimes some articles are in 2 or 3 of them so they are duplicates. How do I remove the duplicates?

**Comment by coolbq** [comment]
If the link of the duplicated articles are the same, I think it will help if you change the link columnin the _entry table to be unique. Then duplicated links will fail.

**Reply by github-yuzer** [reply]
How do I do that?

**Reply by coolbq** [reply]
Maybe phpadmin or Adminer, to change the link field to be unique.

**Comment by Alkarex** [comment]
We have a new feature to automatically mark as read an article for which there is already one with the same title *in the same feed*.
We will soon extend this feature to check for existing titles at category or global levels.

**Reply by github-yuzer** [reply]
Any plans to make this work across feeds if its using the same GUID?

I ask because I'm able to subscribe to feeds of different categories, but those feeds often have the same article marked under different categories. I've yet to find a free reliable feed combiner that removes duplicates, so I have no choice but to subscribe to multiple feeds from the same site. The other problem is some feeds create different feeds under categories, but don't actually put the categories in the feed itself. However the GUID is the same.

**Reply by Alkarex** [reply]
Good idea to also have a GUID option.

**Reply by domalex** [reply]
Is there in the meantime any solution to this problem. I have the same issue. I follow news sites which provide the same articles under different feed categories: latest articles, top articles, international, national, economics, stock markets, sports etc.. As I want to dive into one of those categories selectively, I subscribed to all of the available feeds.

If I read the article in one of those feeds I would like to have it deleted or at least marked as read in all other feeds.

**Reply by Randalix** [reply]
I imported an opml. Now I have a lot of dupliact feeds. How can I remvove thos duplicats?
Is there  a file containing a list of subscribed feeds that I could edit?

**Reply by Peronia** [reply]
Hi @Alkarex, is this feature extended to categories/globat level? In the desc at the feed filter management I can't find something. It's not clear to me. At a category I can't see such filter options.

**Reply by Alkarex** [reply]
> Hi @Alkarex, is this feature extended to categories/globat level?

No, marking as read based on existing titles is only checking titles from the same feed so far. I cannot remember whether we already have an open issue for that, otherwise you are welcome to file a new issue.

**Reply by Peronia** [reply]
Belongs to: https://github.com/FreshRSS/FreshRSS/issues/948

**Comment by Frenzie** [comment]
Is there something specific you'd like to see in subscription management?

**Reply by Randalix** [reply]
I'm looking for a way to remove duplicate feeds.

**Reply by domalex** [reply]
I have 2 features I would like to see in subscription management:

1. Auto remove duplicates

FreshRSS should ideally check for articles which exist in more than one feed, delete copies except of one and create instead a link in the feeds where the article was removed.

Newspapers provide the same articles within different feed collections e.g. latest articles, top articles, international, national, economics, stock markets, sports etc. I usually subscribe to all of these in order to selectively read a particular topic. Consequently I end up with some articles being included in more than one feed. When I read an article in one of the feeds I want the same article to show as read in other collections too.

2. Tags for feeds

In subscription management I would like to mark feeds with one or multiple tags e.g. latest articles, top articles, international, national, economics, stock markets, sports, IT etc. All articles within a feed would inherit the tags. When reading I chose a particular tag and could go through corresponding articles. Duplicates would ideally already be removed as per my suggestion above.

**Reply by Alkarex** [reply]
Automatic tagging is high on my own todo-list, so it will come

**Reply by domalex** [reply]
Thank you for the feedback. That's great news!

**Reply by Alkarex** [reply]
Automatic tagging is ready for testing https://github.com/FreshRSS/FreshRSS/pull/5954

**Reply by domalex** [reply]
I will give it a try. Thank you.

**Reply by domalex** [reply]
During my tests I came across one issue: I am scraping some feeds with the readable plugin. Would it be possible to apply labels only after the scraping process ended? The current solution cannot apply labels which depend on the content of scraped articles.

The same problem I encountered under "subscription management -> filter actions -> mark an article as read...". A filter which is applied on a scraped article doesn't work as labels seem to be applied before the scraped content becomes available.

The same filter under user queries works flawlessly.

**Reply by Alkarex** [reply]
https://github.com/FreshRSS/FreshRSS/issues/6090

**Comment by pax0707** [comment]
Is this in freshrss/freshrss:edge image?

I can see the new label management but no my labels in normal view.

**Reply by Alkarex** [reply]
To see "my labels", there needs to be some tagged unread articles. Alternatively, you can click one the button with the open envelope to show the read articles

**Comment by pax0707** [comment]
User error.
I'm trying to use labels to replicate functionality from user queries and keep mixing which applies to new and which to all.

**Comment by P01NTE45T** [comment]
I'm using my Postgres database for feeds. Would it be feasible to set up a trigger that automatically deletes duplicates based on the article URL each time it's updated? I had a go but it just stopped the entire database from updating. (Thanks in advance.)

**Reply by domalex** [reply]
I support this request. Ideally duplicates would only get deleted after a user tag was added to the remaining article. This way an article could be found under different auto-user-labels: e.g.  latest articles, top articles, international, economics, stock markets, etc.

**Comment by jneumann90** [comment]
Found this very useful SQL-statement as workaround. May be triggered by cron:

`UPDATE entry 
SET is_read = 1
WHERE id NOT IN (
   SELECT MIN(id) 
   FROM entry 
   GROUP BY title
);`

original source (in German):
https://www.tobias-bauer.de/de/blog/2025/2025-07-01-freshrss-duplettenerkennung.html
